File: Programmers/JS/DFSBFS_네트워크.js
function solution(n, computers) {
    var answer = 0;
    var visit = new Array(n);
    
    function networking(idx, computers) {
        for(var l=0; l<n; l++){
            if(computers[idx][l] === 1){
                if(visit[l] !== 1){
                    visit[l] = 1
                    networking(l, computers)
                }
            }
        }
    }
    
    for(var i=0; i < n; i++){
        if (visit[i] !== 1){
            visit[i] = 1
            networking(i, computers)
            answer += 1
        }
    }
    
    return answer;
}

File: Programmers/Python/DFSBFS_네트워크.py
def network(idx, ls):
    global visit
    tg_lst = ls[idx]
    for i in range(len(ls)):
        if visit[i] == 0 and tg_lst[i] == 1:
            visit[i] = 1
            network(i, ls)
            
def solution(n, computers):
    global visit
    answer = 0
    visit = [0] * n
    for a in range(n):
        if visit[a] == 0:
            visit[a] = 1
            network(a, computers)
            answer += 1
    return answer